corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -564,8 +564,12 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
         var self = this;
         self.fromJS(json);
         self.parent = parent;
-        // Grab the parent pubsub so questions can interact with other questions on the same form/group.
-        self.parentPubSub = (parent) ? parent.pubsub : new ko.subscribable();
+        // Grab the parent pubsub so questions can interact with other questions on the same form.
+        let currParent = parent;
+        while (currParent && currParent instanceof Group) {
```

Comment:
I don't think this logic is quite right, and that you want to keep moving up the hierarchy as long as you have a container, not a group. Imagine you have this hierarchy:
```
Form > Repeat > Group
```
You'll end up creating a pubsub on the repeat, not the form.

Reply:
Hi Jenny, sorry I forgot about the Repeat.
I'm wondering, why keep moving up as long as it's a container? I thought we should stop moving up as long as we reach Form. Thus we make sure questions in different forms won't interfere with each other.

Reply:
There's always going to be a single form, and that form will always be the root of the tree. So moving up when you hit a form is also fine, or moving up until you find an item that doesn't have a parent.

[_getDisplayOptions](https://github.com/dimagi/commcare-hq/blob/74d68e4d773a1905413698c7ade32505211a0dd8/corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js#L1167) has similar logic, as it also needs to get the root form. You could consider combining them, that is, making a utility function that takes a question and returns the root form object. (Although, looking at that code, I think 10 might be too small a value for `maxIter`, we have some very deeply nested forms.)

Reply:
Yeah I guess I'll move up until I hit a Form, since Form won't have a parent: https://github.com/dimagi/commcare-hq/blob/c3313d2549560233c8a02650aaa0dd58ba13acb6/corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js#L229
I'll write a utility function for this! Thank you!

Reply:
Hi Jenny, fixed here: ddc7f4ec4233a98235e58fd30f4ccd695d400f9e